src/ck-core/cklocation.ci

```diff
@@ -8,18 +8,17 @@ module CkLocation {
   group [migratable] CkLocCache {
     entry CkLocCache();
     entry [expedited] void requestLocation(CmiUInt8 id, int peToTell);
-    entry [expedited] void updateLocation(CmiUInt8 id, int nowOnPe);
+    entry [expedited] void updateLocation(const CkLocEntry& newEntry);
   };
 
   group [migratable] CkLocMgr {
```

Comment:
More as a future question than something immediately relevant, can CkLocMgr be templated on `Index` and specialized for `CkArrayIndex` by default? What would have to change for this to be possible?

Reply:
I've been wondering something like that myself. There's probably a lot of templating type stuff that can be done within the CkLocMgr/CkLocCache/CkArray/etc space that could be useful. A lot of it is probably pretty straightforward.